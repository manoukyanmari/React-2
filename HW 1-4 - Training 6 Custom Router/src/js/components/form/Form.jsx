// src/js/components/Form.jsx
import React, { Component } from "react";
import { connect } from "react-redux";
import uuidv1 from "uuid";
import { addMovie } from "../../actions/action";
function mapDispatchToProps(dispatch) {
    return {
        addMovie: movie => dispatch(addMovie(movie))
    };
}
// smart component
class ConnectedForm extends Component {
    constructor() {
        super();
        this.state = {
            title: ""
        };
        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }
    handleChange(event) {
        this.setState({ [event.target.id]: event.target.value });
    }
    handleSubmit(event) {
        event.preventDefault();
        const { title } = this.state;
        const id = uuidv1();
        this.props.addMovie({ title, id, src: '../../../src/imgs/machinist.JPG' });
        this.setState({ title: "" });
    }
    render() {
        const { title } = this.state;
        return (
            <form onSubmit={this.handleSubmit}>
                <div className="form-group text-right">
                    <input
                        type="text"
                        className="form-control"
                        id="title"
                        value={title}
                        onChange={this.handleChange}/>
                </div>
                <div className="row">
                    <div className="col-md-6">
                        <span>SEARCH BY: </span>
                        <button type="button" id="title" className="btn btn-primary btn-lg float-left margin-for-btn">
                            TITLE
                        </button>
                        <button type="button" id="title" className="btn btn-primary btn-lg float-left margin-for-btn">
                            GENRE
                        </button>
                    </div>
                    <div className="col-md-6 text-right">
                        <button type="submit" className="btn btn-success btn-lg float-right margin-for-btn">
                            SEARCH
                        </button>
                    </div>
                </div>
            </form>
        );
    }
}
const Form = connect(null, mapDispatchToProps)(ConnectedForm);
export default Form;