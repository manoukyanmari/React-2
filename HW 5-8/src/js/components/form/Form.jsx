// src/js/components/Form.jsx
import React, { Component } from "react";
import { connect } from "react-redux";
import uuidv1 from "uuid";
import store from "../../store/store"
import * as actions from "../../actions/action";
import {Link} from "react-router-dom";
import {corsAJAX} from "../../actions/helpers";



const ROOT_URL = "https://reactjs-cdp.herokuapp.com/";


// smart component
class ConnectedForm extends Component {
    constructor(props) {
        super(props);
        //
        this.state = {
            query: ''
        };
        this.onChangeQuery = this.onChangeQuery.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
        this.setRequest = this.setRequest.bind(this);
    }

    onChangeQuery = (event) => {
        this.setState({query: event.target.value});
       // this.setRequest(event.target.value)
    };

    componentDidMount() {
        const urlParams = new URLSearchParams(window.location.search);
        const q = urlParams.get('query');
        this.state = {
            query: q? q : '',
        };
        if(this.state.query) {
            this.setRequest(q);
        }
    }

    setRequest = (val) => {
        this.setState({query: val});

        fetch(`${ROOT_URL}movies?searchBy=genres&search=${val}`)
            .then(res => res.json())
            .then(
                (result) => {
                    this.setState({
                        articles: result.data
                    });
                },
                (error) => {
                    this.setState({
                        isLoaded: true,
                        error
                    });
                }
            )
    };

    handleSubmit = event => {
        event.preventDefault();
        const { query } = this.state;
        console.log(query, 'dsd');
        this.setRequest(query);

        //
        // //store.dispatch(addArticles(query));
        // corsAJAX(`${ROOT_URL}/search/query=${query}`, 'GET').map(res => res.response.results.filter(content => {
        //     console.log(content, 'sdsd');
        //    // return content
        // }))
        //var title = "New Machinist Query";
        // console.log(this.props, 'dsdsdsd');
        // for (var i = 0; i < 5; i++) {
        //     const id = uuidv1();
        //     this.props.addArticles({  query: query });
        // }
        // console.log(query, 'query');
       // this.setState({ query });
    };
    render() {
        return (<div>
            <form onSubmit={this.handleSubmit}>
                <div className="form-group text-right">
                    <input
                        type="text"
                        className="form-control"
                        id="query"
                        value={this.state.query}
                        onChange={this.onChangeQuery}/>
                </div>
                <div className="row">
                    <div className="col-md-6">
                        <span>SEARCH BY: </span>
                        <button type="button" id="query" className="btn btn-primary btn-lg float-left margin-for-btn">
                            TITLE
                        </button>
                        <button type="button" id="title" className="btn btn-primary btn-lg float-left margin-for-btn">
                            GENRE
                        </button>
                    </div>
                    <div className="col-md-6 text-right">
                        <button type="submit" className="btn btn-success btn-lg float-right margin-for-btn">
                            <Link to={'/movie-list?query=' + this.state.query}>
                                SEARCH
                            </Link>
                        </button>
                    </div>
                </div>
            </form>
                {/*{ this.props.articles.length > 0 ?*/}
                    {/*<Row type={`Results for "${this.state.query}"`}*/}
                         {/*items={this.props.articles} /> : null }*/}
            </div>
        );
    }
}


const mapStateToProps = ({ articles }) => {
    console.log(articles);
    return { articles: articles };
};
function mapDispatchToProps(dispatch) {
    return {
        addArticles: articles => dispatch(addArticles(articles))
    };
}
const Form = connect(mapStateToProps, actions)(ConnectedForm);


export default Form;