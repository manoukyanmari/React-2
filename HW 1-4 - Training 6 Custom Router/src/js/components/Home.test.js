import React from "react";
import {shallow} from "enzyme/build";
import Adapter from 'enzyme-adapter-react-16';
import { configure } from 'enzyme';
import Form from "./form/Form.jsx";
configure({ adapter: new Adapter() });

describe('Home Component',()=>{
    let form;
    let header;

    it('renders correctly the Home Component', () => {
        form = shallow(<Form/>);
    });

    it('renders correctly the Header for home', () => {
        header = shallow(
            <div className="container">
            <div className="jumbotron">
            <h2>NetFlix roulette</h2>
            <label htmlFor="title">FIND YOUR MOVIE</label>
            <Form />
            </div>
            </div>
        );
        expect(header).toMatchSnapshot();
    });
})