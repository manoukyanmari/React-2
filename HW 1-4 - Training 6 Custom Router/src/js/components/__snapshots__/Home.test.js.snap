// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component renders correctly the footer 1`] = `Object {}`;

exports[`Home Component renders correctly the Header for home 1`] = `
<div
  className="container"
>
  <div
    className="jumbotron"
  >
    <h2>
      NetFlix roulette
    </h2>
    <label
      htmlFor="title"
    >
      FIND YOUR MOVIE
    </label>
    <Connect(ConnectedForm) />
  </div>
</div>
`;
